src/Data/Nilai.php

```diff
@@ -79,6 +86,11 @@ public function getSemesters($refetch = false)
         return Enums\Semester::createFromPreg($semester_data);
     }
 
+    /**
```

Comment:
The function will automatically detect if something has been fetched before. Something like auto caching.